### ISSUE TITLE
Piecelists

### ISSUE THREAD

**Issue by ZirconiumX** [japanoise/kusanagi#29]
https://chessprogramming.wikispaces.com/Piece-Lists

**Comment by japanoise** [comment]
This has been on the TODO for a while now.

**Comment by japanoise** [comment]
Implemented.

**Comment by ZirconiumX** [comment]
No, I'm not happy with the current implementation, because we have piecelists for piece -> square conversion, but no square -> piece conversion. I'm going to keep this open for now.

**Comment by ZirconiumX** [comment]
The solution to that would be to expand `board.Data` from `byte` to at least `uint16`, then add an additional bit field to the square.
